### ISSUE TITLE
CMake Error, Can't find Serial package configuration file 

### ISSUE THREAD

**Issue by Xpar65** [joshnewans/diffdrive_arduino#8]
Hi. Whenever I try to build i get the error:
CMake Error at CMakeLists.txt:18 (find_package):
  By not providing "Findserial.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "serial", but
  CMake did not find one.

  Could not find a package configuration file provided by "serial" with any
  of the following names:

    serialConfig.cmake
    serial-config.cmake

  Add the installation prefix of "serial" to CMAKE_PREFIX_PATH or set
  "serial_DIR" to a directory containing one of the above files.  If "serial"
  provides a separate development package or SDK, be sure it has been
  installed.

I have tried:
- sudo apt install libserial-dev
- tried installing from source
- passing the directory of the build files using set(serial_DIR path path/to/serial/build)
- using the name depend>libserial-dev</depend>  instead. Because: https://index.ros.org/d/libserial-dev/
- updating rosdeps. 

Other notes:
- Im using humble
- all other packages are found 

I am not sure how to fix this. My next option is trying to use another serial package instead. Any help would be much appreciated.

**Comment by joshnewans** [comment]
Yeah sorry, that's really not clear if you haven't got the tutorial that went with it!

The serial library is available [here](https://github.com/joshnewans/serial), however you probably want to be using the `humble` branch (`main` is `foxy` and I need to rename it and change the default), and that branch _is_ using `libserial-dev`.

**Comment by eashangallage** [comment]
I have the same issue and when I tried to build the serial lib you have mentioned I got the following issue:

CMake Error at CMakeLists.txt:5 (find_package):
  By not providing "Findcatkin.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "catkin", but
  CMake did not find one.

  Could not find a package configuration file provided by "catkin" with any
  of the following names:

    catkinConfig.cmake
    catkin-config.cmake

  Add the installation prefix of "catkin" to CMAKE_PREFIX_PATH or set
  "catkin_DIR" to a directory containing one of the above files.  If "catkin"
  provides a separate development package or SDK, be sure it has been
  installed.


I am using Ubuntu 22, ROS2 Humble on my developer machine.
